pkg/vault/contracts/Authorizer.sol

```diff
@@ -35,47 +35,84 @@ contract Authorizer is AccessControl, IAuthorizer {
     function canPerform(
         bytes32 actionId,
         address account,
-        address
+        address where
     ) public view override returns (bool) {
-        // This Authorizer ignores the 'where' field completely.
-        return AccessControl.hasRole(actionId, account);
+        return AccessControl.hasRole(actionId, account, where);
     }
 
     /**
-     * @dev Grants multiple roles to a single account.
+     * @dev Grants multiple roles to a single account for a set of contracts.
      */
-    function grantRoles(bytes32[] memory roles, address account) external {
+    function grantRoles(bytes32[] memory roles, address account, address[] calldata where) external {
         for (uint256 i = 0; i < roles.length; i++) {
-            grantRole(roles[i], account);
+            grantRole(roles[i], account, where);
         }
     }
 
     /**
-     * @dev Grants roles to a list of accounts.
+     * @dev Grants multiple roles to a single account for all contracts.
      */
-    function grantRolesToMany(bytes32[] memory roles, address[] memory accounts) external {
+    function grantRolesGlobally(bytes32[] memory roles, address account) external {
+        for (uint256 i = 0; i < roles.length; i++) {
+            grantRoleGlobally(roles[i], account);
+        }
+    }
+
+    /**
+     * @dev Grants roles to a list of accounts for a set of contracts.
+     */
+    function grantRolesToMany(bytes32[] memory roles, address[] memory accounts, address[] calldata where) external {
+        InputHelpers.ensureInputLengthMatch(roles.length, accounts.length);
+        for (uint256 i = 0; i < roles.length; i++) {
+            grantRole(roles[i], accounts[i], where);
+        }
+    }
+
+    /**
+     * @dev Grants roles to a list of accounts for all contracts.
+     */
+    function grantRolesGloballyToMany(bytes32[] memory roles, address[] memory accounts) external {
```

Comment:
These functions are starting to grow a bit out of control: they're fine for now, but I created https://github.com/balancer-labs/balancer-v2-monorepo/issues/986 to propose reducing how many flavors we support.

Reply:
I was not sure about them either but wanted to keep consistent. Also adds some pretty trivial tests and a bit of complexity

Reply:
Makes sense! We can just drop them later as part of #986